.github/workflows/build_macos.yml

```diff
@@ -0,0 +1,122 @@
+name: Build MacOS
+
+on:
+  push:
+    branches:
+      - next
+      - main
+
+env:
+  OCAML_DEFAULT_VERSION: 4.10.0
+
+jobs:
+  install_macos:
+    # Test to install all package on mac with opam
+    name: Install MacOS
+
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Retrieve opam cache
+        uses: actions/cache@v2
+        id: cache-opam
+        with:
+          path: ~/.opam
+          key: v1-${{ runner.os }}-alt-ergo-${{ env.OCAML_DEFAULT_VERSION }}-${{ hashFiles('*.opam') }}
+
+      - name: Use OCaml ${{ env.OCAML_DEFAULT_VERSION }}
+        uses: avsm/setup-ocaml@v1
+        with:
+          ocaml-version: ${{ env.OCAML_DEFAULT_VERSION }}
+
+      - name: Set git user
+        run: |
+          git config --global user.name github-actions
+          git config --global user.email github-actions-bot@users.noreply.github.com
+
+      - name: Pin no actio
+        run: opam pin add . -y --no-action
+
+      - name: Install depext
+        run: opam depext -y alt-ergo-lib alt-ergo-parsers alt-ergo altgr-ergo
+        # if: steps.cache-opam.outputs.cache-hit != 'true'
+
+      - name: Install deps
+        run: opam install -y ./*.opam --deps-only --with-test
+        if: steps.cache-opam.outputs.cache-hit != 'true'
+
+      - name: Fixup
+        run: opam upgrade --fixup
+        if: steps.cache-opam.outputs.cache-hit == 'true'
+
+      - name: Install
+        run: opam install .
+
+  tests_macos:
+    name: Tests MacOS
+
+    needs:
+      - install_macos
+
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Retrieve opam cache
+        uses: actions/cache@v2
+        id: cache-opam
+        with:
+          path: ~/.opam
+          key: v1-${{ runner.os }}-alt-ergo-${{ env.OCAML_DEFAULT_VERSION }}-${{ hashFiles('*.opam') }}
+
+      - name: Use OCaml ${{ env.OCAML_DEFAULT_VERSION }}
+        uses: avsm/setup-ocaml@v1
+        with:
+          ocaml-version: ${{ env.OCAML_DEFAULT_VERSION }}
+
+      - name: Run non regression tests
+        run: opam exec -- rsc/extra/non_regression.sh
```

Comment:
It only re-uses the previous opam cache which already contains the alt-ergo binary. The checkout and ocaml-setup are needed to retrieve the non-regression script, files, and opam (used to run the non-regression script).

At this time no deps or depexts are needed to test this project. I agree with you on the fact that we need to convert this tests to use `dune`. An issue is open (#441) and should be fixed in the futur. Since then, i've added some steps to install specific deps and depexts.